src/resolver.ts

```diff
@@ -7,49 +10,63 @@ export function getRelated(file) {
 }
 
 export function isSpec(file) {
-	return file.indexOf('_spec.rb') > -1;
+	return file.indexOf('_spec.rb') > -1 || file.indexOf('_test.rb') > -1;
+}
+
+function _specDirExists() {
+	try {
+		fs.statSync(path.resolve("./spec")).isDirectory();
+	} catch (e) {
+		if (e.code === 'ENOENT') {
+      return false;
+    } else {
+      throw e;
+    }
+	}
 }
 
 export function codeToSpec(file) {
```

Comment:
`codeToSpec` should be a pure function. So no accessing the file system in there. In this way it will be easily testable.

It can take the `specType`.

```
enum SpecType {
    Rspec,
    MiniTest,
}

export function codeToSpec(file: string, specType: SpecType): string {
```

Then access the file system to determine the spec type just before.  
https://github.com/sporto/rails-go-to-spec-vscode/blob/9b85d11326112148068fb598203c6c517e44af57/src/extension.ts#L46

Something like

```
let specType: SpecType = determineSpecType();
let related: string = resolver.getRelated(fileName, specType);
```

Where `determineSpecType` is a function that access the file system as you are doing.

At least in this way we push the side effects out of that function.